src/core_read.cpp

```diff
@@ -59,6 +60,16 @@ CScript ParseScript(const std::string& s)
         {
             // Number
             int64_t n = atoi64(*w);
+
+            //it's always assumed that a numeric op fits in an int32_t in other code paths,
+            //so enforce an artificial threshold here
+            //for details, see : https://github.com/bitcoin/bitcoin/pull/18413#issuecomment-602954469
+            if(n > std::numeric_limits<int32_t>::max()) {
+                n = std::numeric_limits<int32_t>::max();
+            } else if(n < std::numeric_limits<int32_t>::min()) {
+                n = std::numeric_limits<int32_t>::min();
+            }
```

Comment:
FWIW, here's an alternative shorter version for the same logic, without branches:
```
    n = std::min(n, static_cast<int64_t>(std::numeric_limits<int32_t>::max()));
    n = std::max(n, static_cast<int64_t>(std::numeric_limits<int32_t>::min()));
```
Not sure if it's really more readable though, personally I'd be okay with both versions.

Reply:
`std::clamp` in C++17 would be even better and less verbose, but unsure if we can use C++17 features ?

Note that `min/max` in the standard also do `exactly one comparison` each.

I can remove the `if/else` tho if people think it fits better with the existing codebase.

Reply:
Ah, `std::clamp` would indeed be a very nice option. Unfortunately, upgrading to C++17 has not happened yet (see this discussion: https://github.com/bitcoin/bitcoin/issues/16684), so it can't be used.
Sometimes substitutes for not-yet-available C++ features are implemented by contributors, like `Span` for the C++20 `std::span` by sipa (see https://github.com/bitcoin/bitcoin/pull/12886), don't know though if that is a desired way to go here, and if yes, if it should be packed in this PR. This should probably include identifiying all clamps throughout the codebase and using our newly introduced clamp function (so it can be replaced with `std::clamp` once we move to C++17).

Reply:
Thanks, I'd be down to do that in a different PR. I'm looking for stuffs to do to improve my knowledge of the codebase anyway :)

I'll wait for some other people to confirm "yes this is a good idea, do a separate PR for std::clamp" before starting though